x-pack/plugins/alerts/server/saved_objects/migrations.ts

```diff
@@ -12,53 +12,48 @@ import {
 import { RawAlert } from '../types';
 import { EncryptedSavedObjectsPluginSetup } from '../../../encrypted_saved_objects/server';
 
+export const LEGACY_LAST_MODIFIED_VERSION = 'pre-7.10.0';
+
 export function getMigrations(
   encryptedSavedObjects: EncryptedSavedObjectsPluginSetup
 ): SavedObjectMigrationMap {
-  const alertsMigration = changeAlertingConsumer(encryptedSavedObjects, 'alerting', 'alerts');
-
-  const infrastructureMigration = changeAlertingConsumer(
-    encryptedSavedObjects,
-    'metrics',
-    'infrastructure'
-  );
+  const migrationWhenRBACWasIntroduced = markAsLegacyAndChangeConsumer(encryptedSavedObjects);
 
   return {
-    '7.10.0': (doc: SavedObjectUnsanitizedDoc<RawAlert>, context: SavedObjectMigrationContext) => {
-      if (doc.attributes.consumer === 'alerting') {
-        return executeMigration(doc, context, alertsMigration);
-      } else if (doc.attributes.consumer === 'metrics') {
-        return executeMigration(doc, context, infrastructureMigration);
-      }
-      return doc;
-    },
+    '7.10.0': executeMigrationWithErrorHandling(migrationWhenRBACWasIntroduced),
   };
 }
 
-function executeMigration(
-  doc: SavedObjectUnsanitizedDoc<RawAlert>,
-  context: SavedObjectMigrationContext,
+function executeMigrationWithErrorHandling(
   migrationFunc: SavedObjectMigrationFn<RawAlert, RawAlert>
 ) {
-  try {
-    return migrationFunc(doc, context);
-  } catch (ex) {
-    context.log.error(
-      `encryptedSavedObject migration failed for alert ${doc.id} with error: ${ex.message}`,
-      { alertDocument: doc }
-    );
-  }
-  return doc;
+  return (doc: SavedObjectUnsanitizedDoc<RawAlert>, context: SavedObjectMigrationContext) => {
+    try {
+      return migrationFunc(doc, context);
+    } catch (ex) {
+      context.log.error(
+        `encryptedSavedObject migration failed for alert ${doc.id} with error: ${ex.message}`,
```

Comment:
**nit**: This isn't directly related to this PR but I noticed it just now, especially that the migration will run on every alert. The error message is moving further away from knowing ESO is the cause of the error, it would be nice for it to list the migration version instead of `encryptedSavedObject` in the message.

Reply:
👍